### PR TITLE
Added gitleaks to precommit

### LINKED ISSUE #53: Add `gitleaks` to `pre-commit`
https://github.com/gitleaks/gitleaks?tab=readme-ov-file#pre-commit

### PR DESCRIPTION
**Pull request by Virtual4087**
Closes #53

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -43,3 +43,8 @@ repos:
     hooks:
       - id: codespell
         args: ["--ignore-words", ".github/linters/codespell.txt"]
+
+  - repo: https://github.com/gitleaks/gitleaks
+    rev: v8.19.0
```
Comment (jbampton) [inline .pre-commit-config.yaml:48]:
```suggestion
    rev: v8.21.2
```

**Review by jbampton — APPROVED** [review]

**Comment by jbampton** [comment]
Thank you :+1:

**Comment by jbampton** [comment]
Hey @Virtual4087 I sent you an invite to join the Brisbane Social Chess GitHub Organization.

It is safe to join and we invite all contributors.

Thanks once again.

**Comment by Virtual4087** [comment]
@jbampton Thanks for the invitation. I'll join it.

**Comment by jbampton** [comment]
Thanks for joining @Virtual4087 

You can set your membership in the Brisbane Social Chess GitHub Organization to public on the people page when logged in:

https://github.com/orgs/brisbanesocialchess/people

Please show your support for your new team :)

If you are a private member, the public including everyone else that is not in the team won't see that you are part of Brisbane Social Chess.

Your choice.  Thanks again.

**Comment by Virtual4087** [comment]
@jbampton Got it, I've set it to public.

**Pull request merged**